src/charts/scatter-plot.js

```diff
@@ -480,6 +542,59 @@ define(function(require) {
             }
         }
 
+        /**
+         * Calculates and returns
+         * @return {Object}
+         * @param {*} svg
+         * @private
+         */
+        function getPointProps(svg) {
+            let mousePos = d3Selection.mouse(svg);
+            mousePos[0] -= margin.left;
+            mousePos[1] -= margin.top;
+
+            return {
+                closestPoint: voronoi.find(mousePos[0], mousePos[1]),
+                mousePos
+            };
+        }
+
+        /**
+         * Handler called on mousemove event
+         * @return {void}
+         * @private
+         */
+        function handleMouseMove(svg, d) {
+            let { mousePos, closestPoint } = getPointProps(svg);
+
+            tooltip.update(getPointData(closestPoint), mousePos, [chartWidth, chartHeight]);
```

Comment:
Usually we use the Tooltip as an external chart, could we do the same here?

Reply:
I guess we would pass the 'getPointData(closestPoint)' instead of 'd' down in the dispatcher call.